core/project/lib/deprecate-config.js

```diff
@@ -36,12 +41,17 @@ function remap(search, target, { alsoRoot } = {}) {
       if (dotProp.has(obj.config, searchPath)) {
         const localPath = path.relative(".", obj.filepath);
 
+        const fromVal = dotProp.get(obj.config, searchPath);
+        const toVal = toValue ? toValue(fromVal) : fromVal;
+        const from = stringify({ [searchPath]: fromVal });
+        const to = stringify({ [target]: toVal });
+
         log.warn(
           "project",
-          `Deprecated key "${searchPath}" found in ${localPath}\nPlease rename "${searchPath}" => "${target}"`
+          `Deprecated key "${searchPath}" found in ${localPath}\nPlease update ${from} => ${to}`
```

Comment:
I think you could factor out the update message generation into a method of its own, it would be clearer. Also, if the values haven't changed, we should probably omit them from the message, it's just confusing.